lib/pulsar/broker.ex

```diff
@@ -701,6 +721,22 @@ defmodule Pulsar.Broker do
     end
   end
 
+  defp handle_command(%Binary.CommandProducerSuccess{} = command, broker) do
+    # CommandProducerSuccess can arrive twice for WaitForExclusive mode:
+    # 1. First with producer_ready: false (pending state) - request_id in broker.requests
+    # 2. Second with producer_ready: true (final state) - request_id NOT in broker.requests (already cleaned up)
+    request_id = command.request_id
+
+    # Check if this request_id is still pending in broker.requests
+    if Map.has_key?(broker.requests, request_id) do
+      # Initial registration response - request is still pending
+      handle_producer_registration_response(command, broker, request_id)
+    else
+      # Subsequent notification - request was already completed
+      handle_producer_ready_notification(command, broker)
```

Comment:
Depending on how we go about the above, this could also be a simple `reply_to_request/3` call? We would, however, need to adjust `reply_to_request/3` to fallback to send the response to a matching consumer/producer if the request id cannot be found. What do you think?